src/build/app/mod.rs

```diff
@@ -1572,15 +1573,14 @@ impl<'help> App<'help> {
     /// ```rust
     /// # use clap::App;
     /// use std::io;
-    /// let mut app = App::new("myprog");
-    /// let mut out = io::stdout();
-    /// app.write_long_version(&mut out).expect("failed to write to stdout");
+    /// let app = App::new("myprog");
+    /// println!("{}", app.render_long_version());
     /// ```
     /// [`io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
     /// [`-V` (short)]: ./struct.App.html#method.version
     /// [`--version` (long)]: ./struct.App.html#method.long_version
-    pub fn write_long_version<W: Write>(&self, w: &mut W) -> ClapResult<()> {
-        self._write_version(w, true).map_err(From::from)
+    pub fn render_long_version(&self) -> String {
```

Comment:
Hmm.. I understand what we are trying to do here. But can you please comment with a sentence or two on why we don't want `write_` fns anymore?

Reply:
Damn, I wish I could remember... 

I vaguely recall that was something about weird borrowcheck errors involving the `Write` here. I don't really think there's much point to keep this function around since we now have `get_version`. It doesn't do much anyway.